src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -32,7 +33,7 @@ public IEnumerable<string> Scopes
         {
             get
             {
-                if (_scopes != null)
+                if (_scopes is object)
```

Comment:
I have not seen this type of check before. I am curious - how does it differ?

Reply:
This is a new feature in [C# 7.0](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7#pattern-matching), part of added pattern matching. It bypasses any potential operator overloads (for either `==` and/or `!=`) and directly emits the intended IL for null/not null comparison.

Reply:
nit: While this is definitely interesting, these checks obfuscate the fact that we are checking for null here. The word "is" usually implies that you are checking if an object is of a certain type, not whether or not it is null so this may be confusing to someone who is not aware of this feature and they may end up adding the null check in addition to this.

Reply:
I agree that it's a (relatively) new feature and might be confusing. As any other new feature :)

There are some potential benefits in using it, plus many most popular OSS projects have switched to using it too.

So here's our options:
1. Continue to use `!= null` and `== null`
2. Use `is object` and `is null` (my preference)
3. Use `is not null` and `is null` (I personally find the former the least readable)

Reply:
I like option 1 or 3.
Thoughts @bgavrilMS?

Reply:
Very interesting problem. It looks like "obj is null" is something we can do today. But "object is not null" is a C# 9.0 feature, which we don't currently use (but we probably should). 

I agree that "a is object" is a bit counter-intuitive.

According to https://stackoverflow.com/questions/40676426/what-is-the-difference-between-x-is-null-and-x-null it looks like the compiler (Roslyn) was actually modified so that it emits the same kind of logic as long as `==` and `!=` are not overloaded (I believe we haven't overloaded them for any object in MSAL). '

So maybe let's keep on using option 1 for consistency with MSAL, and separately we can look at moving to C# 9 and adding some analyzers for this kind of problem. Any changes should be consistent across the lib.

Reply:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2921